check_all_options.c

```diff
@@ -101,6 +101,30 @@ void check_options_tcp_rr(struct options *opts, struct callbacks *cb)
 
 void check_options_tcp_stream(struct options *opts, struct callbacks *cb)
 {
+#ifdef WITH_TCPDIRECT
+      if (opts->tcpd_gpu_pci_addr) {
+            CHECK(cb, opts->tcpd_nic_pci_addr,
+                  "Must provide NIC PCI address if GPU PCI address was provided.");
+            CHECK(cb, opts->tcpdirect_phys_len > 0,
+                  "Must provide non-zero --tcpdirect-phys-len flag if GPU PCI address was provided.");
+            // TODO check page-alignment
+            // CHECK((CUdeviceptr)gpu_gen_mem_ % PAGE_SIZE == 0);
+
+            if (opts->client) {
+                  CHECK(cb, !opts->tcpd_rx_cpy,
+                        "Copying CUDA buffer to userspace only allowed on hosts.");
```

Comment:
I don't understand what 'hosts' here refers to.

Reply:
receivers (as opposed to clients/senders)